### DISCUSSION TITLE
Custom script

### DISCUSSION THREAD

**Discussion by andershagman** [netbox-community/netbox#7496] [Q&A] [answered]
I have a custom script that run on Netbox ver 2.7. Now on 2.11.12 it does not run. Stops in pendling.
I get the form and can fill it out but after clicking "Run Script"  just ... for ever. 

![bild](https://user-images.githubusercontent.com/20284974/136506737-31fa13cf-c4df-4525-8dbd-981440b63e5a.png)

How do I troubleshoot custom scripts? I have added script logging to configuration.py for scripts but get nothing about this.
Logging seams to work as I get a log rows when removing a script result.

I have copied the example from documentation and that also stops at pending...

**Comment by candlerb** [comment, marked as answer]
> I have copied the example from documentation and that also stops at pending...

My first guess is that it could be something to do with Redis.  The release notes for Netbox 2.9 say:

"When running a report or custom script, its execution is now queued for background processing and the user receives an immediate response indicating its status. This prevents long-running scripts from resulting in a timeout error. Once the execution has completed, the page will automatically refresh to display its results."

Check that netbox-rq is running, and whether anything is being logged there:

```
systemctl status netbox-rq
journalctl -eu netbox-rq
```

Also compare your configuration.py with configuration.example.py, and check if anything is missing.

```
cd /opt/netbox/netbox/netbox
diff -u configuration.py configuration.example.py | less
```

**Reply by andershagman** [reply]
Thanks
The rq-worker wasn't running. Some kind of problem stoped it.

![bild](https://user-images.githubusercontent.com/20284974/136547188-792dc344-502d-45e0-beaf-494ce27ebe38.png)

**Reply by candlerb** [reply]
Glad you found it.  "Error 111 connecting to 127.0.0.1:6379. Connection refused" means that Redis wasn't running at the time that the netbox-rq service started.

If you are using the standard `netbox-rq.service` from the contrib directory then it should automatically try to [restart every 30 seconds](https://github.com/netbox-community/netbox/blob/develop/contrib/netbox-rq.service#L16-L17).

**Reply by andershagman** [reply]
I have "Restart=on-failure". Isn't that good enough.

**Reply by candlerb** [reply]
If it's under the `[Service]` section then it should be.  If you change this file then you need to do `systemctl daemon-reload`.

If you don't override `RestartSec` then it will restart every 100 milliseconds by default!

**Reply by andershagman** [reply]
OK, thanks.
I'm not a systemd-guy, well not a Linux-guy either.

/Anders
The BSD-guy
